### PR TITLE
ET-301 - Update LMS - create mechanism to track users course usage

### PR DESCRIPTION
**Pull request by dmitry-viskov**
Part of ET-100 (As a Librarian, I want to see Modules usage (sessions) by selected time frames)

### CODE REVIEW

**Comment by dpowell** [comment]
Let's include the first and last touch timestamps in the database record. We don't immediately plan to use them, but I suspect might be helpful and seems easy to capture.

**Comment by dmitry-viskov** [comment]
@dpowell PR updated

**Review by dpowell — APPROVED** [review]

File: common/djangoapps/credo_modules/middleware.py [hunk 1]
```diff
@@ -11,3 +17,45 @@ def process_response(self, request, response):
...
+
+        if request.user.is_authenticated() and len(path_data) > 2:
+            course_id = path_data[2]
+
+            course_usage_cookie = request.COOKIES.get(self.course_usage_cookie, '')
+            if course_usage_cookie:
+                try:
+                    course_usage_cookie = json.loads(course_usage_cookie)
+                except ValueError:
+                    pass
```
Comment (snowcar) [inline common/djangoapps/credo_modules/middleware.py]: In this case _course_usage_cookie_ will contain incorrect json string, and the next logic will be corrupted. 
For example we could use `course_usage_cookie=None`

Comment (snowcar) [reply common/djangoapps/credo_modules/middleware.py]: course_usage_cookie = request.COOKIES.get(self.course_usage_cookie, '')
              try:
                   course_usage_cookie_dict = json.loads(course_usage_cookie)
               except ValueError:
                   course_usage_cookie_dict = {}


I think if we init `course_usage_cookie_dict` under try/catch we could avoid additional logic 

     course_usage_cookie_dict = course_usage_cookie if course_usage_cookie else {}

**Review by snowcar — APPROVED** [review]

**Pull request merged**